#### HARDWARE CONFIGURATIONS ####

# BEGIN SENSOR CONFIGS
hardware:
  can_interface: ""

  sensor:
    uwb_node_1:
      id: 1
      name: 'UWB NODE 1'
      type: 'uwb'
      can_id: 1
      update_period: 0.1
    uwb_node_2:
      id: 2
      name: 'UWB NODE 2'
      type: 'uwb'
      can_id: 2
      update_period: 0.1
    uwb_node_3:
      id: 3
      name: 'UWB NODE 3'
      type: 'uwb'
      can_id: 3
      update_period: 0.1
    uwb_node_4:
      id: 4
      name: 'UWB NODE 4'
      type: 'uwb'
      can_id: 4
      update_period: 0.1
    ebay_temperature:
      id: 5
      name: 'EBAY TEMPERATURE'
      gpio: '/sys/class/gpio/gpio69/'
      over_temp_gpio: '/sys/class/gpio/gpio27/'
      critical_temp_gpio: '/sys/class/gpio/gpio61/'
      update_period: 1.0
    imu:
      id: 6
      name: 'IMU'
      type: 'imu'
      gpio: '/sys/class/gpio/gpio66/'
      update_period: 0.005
    adc_1:
      id: 7
      name: 'POT'
      type: 'pot'
      gpio: '/sys/class/gpio/gpio67/'
      update_period: 0.01
    adc_2:
      id: 8
      name: 'LOAD CELL'
      type: 'load_cell'
      gpio: '/sys/class/gpio/gpio68/'
      update_period: 0.01
    limit_1:
      id: 9
      name: 'LIMIT DEP LOWER'
      type: 'limit'
      gpio: '/sys/class/gpio/gpio60/'
      topic: '/dumper/dep_bottom_limit_switch'
      motor_id: 2
      allowed_dir: 1
    limit_2:
      id: 10
      name: 'LIMIT DEP UPPER'
      type: 'limit'
      gpio: '/sys/class/gpio/gpio48/'
      topic: '/dumper/top_limit_switch'
      motor_id: 2
      allowed_dir: 1
    limit_3:
      id: 11
      name: 'LIMIT EXC TRANS LOWER'
      type: 'limit'
      gpio: '/sys/class/gpio/gpio49/'
      topic: '/excavation/exc_lower_limit_switch'
      motor_id: 2
      allowed_dir: 1
    limit_4:
      id: 12
      name: 'LIMIT EXC TRANS UPPER'
      type: 'limit'
      gpio: '/sys/class/gpio/gpio117/'
      topic: '/excavation/exc_upper_limit_switch'
      motor_id: 2
      allowed_dir: 1
    estop:
      id: 13
      name: "ESTOP"
      type: "estop"
      gpio: "/sys/class/gpio/gpio45"

# BEGIN MOTOR CONFIGS
  motor:
    port_drive:
      id: 5
      name: 'PORT DRIVE'
      type: 'vesc'
      can_id: 5
      gear_reduction: 630 # gear_reduc*pole_pairs
      max_rpm: 100
      max_acceleration: 75
      update_period: 0.25
    starboard_drive:
      id: 6
      name: 'STARBOARD DRIVE'
      type: 'vesc'
      can_id: 6
      gear_reduction: 630 # gear_reduc*pole_pairs
      max_rpm: 100
      max_acceleration: 75
      update_period: 0.25
    dep:
      id: 7
      name: 'DEPOSITION DUNKERINO'
      type: 'vesc'
      can_id: 7
      gear_reduction: 245 # gear_reduc*pole_pairs
      max_rpm: 140
      max_acceleration: 50
      update_period: 0.25
    exc_belt:
      id: 8
      name: 'EXCAVATION BELT'
      type: 'vesc'
      can_id: 8
      gear_reduction: 245 # gear_reduc*pole_pairs
      max_rpm: 140
      max_acceleration: 50
      update_period: 0.25
    exc_translation:
      id: 9
      name: 'EXCAVATION TRANSLATION'
      type: 'vesc'
      can_id: 9
      gear_reduction: 245 # gear_reduc*pole_pairs
      max_rpm: 140
      max_acceleration: 50
      update_period: 0.25
    exc_port_act:
      id: 10
      name: 'EXCAVATION PORT ACTUATOR'
      type: 'brushed_motor'
      can_id: 10
    exc_starboard_act:
      id: 11
      name: 'EXCAVATION STARBOARD ACTUATOR'
      type: 'brushed_motor'
      can_id: 11
